tools/ci_build/provider_ops_disabler.py

```diff
@@ -0,0 +1,381 @@
+'''rewrite execution providers to disable ops'''
+# !/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+import argparse
+import shutil
+import onnx
+from onnx import AttributeProto as AP
+from logger import log
+
+
+domain_map = {'': 'kOnnxDomain',
+              'ai.onnx': 'kOnnxDomain',
+              'ai.onnx.ml': 'kMLDomain',
+              'com.microsoft': 'kMSDomain',
+              'com.microsoft.nchwc': 'kMSNchwcDomain',
+              'com.microsoft.mlfeaturizers': 'kMSFeaturizersDomain',
+              'com.microsoft.dml': 'kMSDmlDomain',
+              'com.intel.ai': 'kNGraphDomain',
+              'com.xilinx': 'kVitisAIDomain'}
+
+
+def map_domain(domain):
+
+    if domain in domain_map:
+        return domain_map[domain]
+
+    return 'UnknownDomain'
+
+
+def extract_ops_from_file(file_path, referred_ops):
+    '''extract ops from file of format: domain;opset;op1,op2...'''
+
+    if not file_path:
+        return referred_ops
+
+    if not os.path.isfile(file_path):
+        log.warning('File {} does not exist'.format(file_path))
+        return referred_ops
+
+    with open(file_path, 'r') as file_to_read:
+
+        for stripped_line in [line.strip() for line in
+                              file_to_read.readlines()]:
+
+            if stripped_line.startswith("#"):  # skip comments
+                continue
+
+            raw_domain, opset, raw_ops =\
+                [segment.strip() for segment in stripped_line.split(';')]
+
+            domain = map_domain(raw_domain)
+            operators = list(set([raw_op.strip() for raw_op in raw_ops.split(',')]))
+
```

Comment:
By converting back to a list you still get duplicates across models don't you? I was suggesting using a set() for the referred_ops[domain][opset] so that does not happen.

Reply:
Done.